frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/ApplicationMaster.java

```diff
@@ -335,6 +336,8 @@ private ContainerRequest setupContainerRequest(TaskStatus taskStatus) throws Exc
     String requestNodeGpuType = requestManager.getTaskPlatParams().get(taskRoleName).getTaskNodeGpuType();
     ResourceDescriptor requestResource = requestManager.getTaskResources().get(taskRoleName);
     ResourceDescriptor maxResource = conf.getMaxResource();
+    Integer taskNubmer = requestManager.getTaskRoles().get(taskRoleName).getTaskNumber();
+    List<Range> preSelectedPorts = statusManager.getTaskRolePortRanges(taskRoleName);
```

Comment:
How do you timeout the preSelectedPorts and use another "preSelectedPorts"? You may need some cleanup logic for the preSelectedPorts.